cc/tf_test.cc

```diff
@@ -0,0 +1,98 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <vector>
+
+#include "tensorflow/core/framework/graph.pb.h"
+#include "tensorflow/core/lib/core/errors.h"
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/core/public/session.h"
+
+#include "cc/constants.h"
+
+using tensorflow::DT_FLOAT;
+using tensorflow::Env;
+using tensorflow::GraphDef;
+using tensorflow::NewSession;
+using tensorflow::ReadBinaryProto;
+using tensorflow::Session;
+using tensorflow::SessionOptions;
+using tensorflow::Status;
+using tensorflow::string;
+using tensorflow::Tensor;
+using tensorflow::TensorShape;
+
+Status DoIt() {
+  GraphDef graph_def;
+  TF_RETURN_IF_ERROR(
+      ReadBinaryProto(Env::Default(), "/tmp/graph/test.pb", &graph_def));
+
+  std::unique_ptr<Session> session(NewSession(SessionOptions()));
+  TF_RETURN_IF_ERROR(session->Create(graph_def));
+
+  Tensor input(DT_FLOAT, TensorShape({1, minigo::kN, minigo::kN, 17}));
+  auto x = input.tensor<float, 4>();
+  for (int j = 0; j < minigo::kN; ++j) {
+    for (int i = 0; i < minigo::kN; ++i) {
+      x(0, j, i, 16) = 1;
+    }
+  }
+  for (int j = 0; j < minigo::kN; ++j) {
```

Comment:
if test code is supposed to serve as a usage example, then some variable names/comments here would be very needed.

Reply:
This is just my first attempt at getting a model to work with the tensorflow C++ API and is not really meant for public consumption. I'll likely delete this whole file once the MCTS player is working.